quantum/secure.c

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 QMK
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "secure.h"
+#include "timer.h"
+
+#ifndef SECURE_UNLOCK_TIMEOUT
+#    define SECURE_UNLOCK_TIMEOUT 5000
+#endif
+
+#ifndef SECURE_IDLE_TIMEOUT
+#    define SECURE_IDLE_TIMEOUT 60000
+#endif
+
+#ifndef SECURE_UNLOCK_SEQUENCE
+#    define SECURE_UNLOCK_SEQUENCE \
+        {                          \
+            { 0, 0 }               \
+        }
+#endif
+
+secure_status_t secure_status = SECURE_LOCKED;
```

Comment:
```suggestion
static secure_status_t secure_status = SECURE_LOCKED;
```

Should this become private to prevent things like this in keyboards and force the use of the public API?

```c
#include "secure.h"

extern secure_status_t secure_status;

void housekeeping_task_user(void) {
    secure_status = SECURE_UNLOCKED; 
}
```

Reply:
It wasnt static because it was originally being used directly by the XAP codegen, though I do need to go double check if that is still the case.